packages/loopring_v3/contracts/core/impl/libtransactions/WithdrawTransaction.sol

```diff
@@ -79,20 +85,29 @@ library WithdrawTransaction
         Withdrawal memory withdrawal = readWithdrawal(data, offset);
         WithdrawalAuxiliaryData memory auxData = abi.decode(auxiliaryData, (WithdrawalAuxiliaryData));
 
-        // Validate the auxixliary withdrawal data
-        if (withdrawal.dataHash == 0) {
-            require(auxData.auxiliaryData.length == 0, "AUXILIARY_DATA_NOT_ALLOWED");
-        } else {
-            // Hashes are stored using only 253 bits so the value fits inside a SNARK field element.
-            require(
-                uint(keccak256(auxData.auxiliaryData)) >> 3 == uint(withdrawal.dataHash),
-                "INVALID_WITHDRAWAL_AUX_DATA"
-            );
-        }
+        // Validate the withdrawal data not directly part of the DA
+        bytes32 dataHash = keccak256(
+            abi.encodePacked(
+                auxData.minGas,
```

Comment:
maybe we should also include validUntil here.

Reply:
It's not necessary because the hash is used for approving the data in the circuit with EdDSA. And when EdDSA signatures are used the timestamp is checked in the circuit and `validUntil` is directly signed by the user.

When not using EdDSA the dataHash is actually not used, because all that data that is not part of the DA is approved directly by the user using an ECDSA signature/approval which directly contains `validUntil`/`minGas`/`data`/`to` as parameters of the onchain hash.